src/core/project/qgsproject.h

```diff
@@ -1362,6 +1362,24 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
      */
     bool removeAttachedFile( const QString &path );
 
+    /**
+     * Returns an identifier for an attachment file path
+     * An attachment identifier is a string which is does not depend on the project archive
+     * storage location.
+     * \param attachedFile An attachment file path
+     * \return An identifier for the attached file
+     * \since QGIS 3.22
+     */
+    QString getAttachmentIdentifier( const QString &attachedFile ) const;
```

Comment:
I'm wondering if these functions wouldn't be better in a "QgsProjectUtils" class, just to keep the already heavy QgsProject api minimized. Thoughts?

Reply:
I think they are ok where they are now, it improves discoverability in the api docs, code completion etc.
Is the `get` prefix required/needed here?